### PR TITLE
feat(theme): support page layout (close: #45)

### PR DESCRIPTION
**Pull request by ulivz**
close: #45

### CODE REVIEW

File: packages/vuepress-theme-vt/layouts/Layout.vue [hunk 1]
```diff
@@ -154,7 +182,20 @@ export default {
           this.toggleSidebar(false)
         }
       }
-    }
-  }
-}
+    },
+
+    getPageLayout() {
+      if (this.$page.path) {
```
Comment (spencerHT) [inline packages/vuepress-theme-vt/layouts/Layout.vue:188]: What does this `this.$page.path` mean ?

File: packages/vuepress-theme-vt/layouts/Layout.vue [hunk 2]
```diff
@@ -92,16 +97,32 @@ export default {
...
         && frontmatter.sidebar !== false
         && this.sidebarItems.length
       )
     },
 
+    shouldShowToc() {
+      const { frontmatter } = this.$page
+      return (
+        this.showShowSideSection
+        && frontmatter.toc !== false
```
Comment (spencerHT) [inline packages/vuepress-theme-vt/layouts/Layout.vue:122]: It looks like add new frontmatter variables `pageLayout ` and `toc`.  
Does it need to add into `packages/vuepress-docs/docs/guide/frontmatter.md`?

Comment (ulivz) [reply packages/vuepress-theme-vt/layouts/Layout.vue:122]: It has been documented but not implemented:

https://vuepress-theme-vt.vercel.app/api/config-frontmatter.html#toc

File: packages/vuepress-theme-vt/layouts/Layout.vue (hunk 1 above)
Comment (ulivz) [reply packages/vuepress-theme-vt/layouts/Layout.vue:188]: Copied from VuePress but I forgot the motivation (so long ago), maybe I can remove it.

**Review by spencerHT — APPROVED** [review]
LGTM

**Pull request merged**